lib/controller.dart

```diff
@@ -268,15 +305,45 @@ void _computeFilterDropdownsAndRender() {
 void _computeChartBucketsAndRender() {
   var charts = _config.tabs[_selectedAnalysisTabIndex].charts;
   _computeChartBuckets(charts);
-  for (var chart in charts) {
-    switch (chart.type) {
+
+  for (var i = 0; i < charts.length; ++i) {
+    var chart = charts[i];
+    switch (charts[i].type) {
       case model.ChartType.bar:
         view.renderBarChart(
             chart.title,
             chart.narrative,
             chart_helper.generateBarChartConfig(chart, _dataComparisonEnabled,
                 _activeFilterValues, _activeComparisonFilterValues));
         break;
+      case model.ChartType.map:
+        var mapData =
+            _mapsGeoJSON[chart.geography.country][chart.geography.regionLevel];
+        var mapValues = Map<String, List<num>>();
+        var mapComparisonValues = Map<String, List<num>>();
+        for (var field in chart.fields) {
+          var regionName = field.field.value.toString();
+          mapValues[regionName] = [
+            field.bucket[0],
+            field.bucket[0] / _filterValuesCount,
+          ];
+          mapComparisonValues[regionName] = [
+            field.bucket[1],
+            field.bucket[1] / _comparisonFilterValuesCount
+          ];
+        }
+
+        view.renderGeoMap(
+          i.toString(),
```

Comment:
Why does `renderGeoMap` need an id, whereas `renderBarChart` doesn't?

Reply:
The `renderBarchart` just takes a chart config which is container independent. 

The `renderGeoMap` needs a container (id) that is unique across all maps in the tab. From what I understand, mapbox library expects a container to be present in the dom and mapbox's render function takes the container's id. So I will have to make a unique ID for each of the mapbox's container.